### PR TITLE
587 show timeseries when double clicking layer

### LINKED ISSUE #587: Show timeseries when double clicking layer
Previously handled in #214

### PR DESCRIPTION
**Pull request by hvangeffen**
Closes #587 

Docs say convertDatum is present on timeseriesGridActions:
https://fewsdocs.deltares.nl/webservices/rest-api/v1/#get-/timeseries/grid/actions

But request fails:
https://rwsos-dataservices-ont.avi.deltares.nl/wqps/FewsWebServices/rest/fewspiservice/v1/timeseries/grid/actions?layers=oceanographic_currents_d3d&x=6363578.691197484&y=1981090.8619554942&startTime=2024-01-17T14%3A00%3A00Z&endTime=2024-01-23T06%3A00%3A00Z&bbox=779235.8504813%2C96197.66808258%2C8571601.37615368%2C5940472.16265769&documentFormat=PI_JSON&elevation=-1&useDisplayUnits=true&convertDatum=false

### CODE REVIEW

File: src/components/spatialdisplay/SpatialDisplay.vue [hunk 1]
```diff
@@ -41,15 +58,83 @@ const route = useRoute()
...
+    y,
+    startTime,
+    endTime,
+    bbox,
+    documentFormat: 'PI_JSON',
+    elevation: elevation.value ?? layerCapabilities.value.elevation?.lowerValue,
+  }
+}
+
+const options = computed<UseDisplayConfigOptions>(() => {
```
Comment (amarkensteijn) [inline src/components/spatialdisplay/SpatialDisplay.vue:103]: `options` is too generic. Rename to `displayConfigOptions` or to `filterOptions`.

File: src/components/spatialdisplay/SpatialDisplay.vue [hunk 2]
```diff
@@ -73,6 +159,38 @@ function openLocationTimeSeriesDisplay(locationId: string) {
...
+    +event.lngLat.lat.toFixed(3),
+    +event.lngLat.lng.toFixed(3),
+  )
+}
+
+function openCoordinatesTimeSeriesDisplay(latitude: number, longitude: number) {
+  const routeName = route.name
+    ?.toString()
+    .replace('SpatialDisplay', 'SpatialTimeSeriesDisplay')
+    .replace('WithCoordinates', '')
```
Comment (amarkensteijn) [inline src/components/spatialdisplay/SpatialDisplay.vue:175]: Is redundant?

File: src/components/spatialdisplay/SpatialDisplayComponent.vue [hunk 3]
```diff
@@ -33,20 +41,22 @@ import MapComponent from '@/components/map/MapComponent.vue'
...
 import LocationsLayer from '@/components/wms/LocationsLayer.vue'
+import SelectedCoordinateLayer from '@/components/wms/SelectedCoordinateLayer.vue'
 import ElevationSlider from '@/components/wms/ElevationSlider.vue'
 import DateTimeSlider from '@/components/general/DateTimeSlider.vue'
 import { DateController } from '@/lib/TimeControl/DateController.ts'
 import debounce from 'lodash-es/debounce'
 import { useUserSettingsStore } from '@/stores/userSettings'
 import type { MapLayerMouseEvent, MapLayerTouchEvent } from 'mapbox-gl'
+import { configManager } from '@/services/application-config'
+import { Layer } from '@deltares/fews-wms-requests'
```
Comment (amarkensteijn) [inline src/components/spatialdisplay/SpatialDisplayComponent.vue:59]:
```suggestion
import type { Layer } from '@deltares/fews-wms-requests'
```

**Review by amarkensteijn — COMMENTED** [review]
It doesn't work for the `SpatialDisplayView`, but that should also be supported.

**Comment by hvangeffen** [comment]
> It doesn't work for the `SpatialDisplayView`, but that should also be supported.

Added to spatialdisplayview

**Pull request merged**